modules/rules_rust/0.34.1/MODULE.bazel

```diff
@@ -0,0 +1,144 @@
+"""bazelbuild/rules_rust"""
+
+module(
+    name = "rules_rust",
+    version = "0.34.1",
+)
+
+print("WARNING: The rules_rust Bazel module is still highly experimental and subject to change at any time. Only use it to try out bzlmod for now.")  # buildifier: disable=print
```

Comment:
`print`s in module files are displayed when the module file is evaluated, which happens during version resolution and thus even if this module doesn't end up being selected. For this reason, getting rid of this warning is extremely difficult as you need to make sure its never mentioned by any module file in the full transitive closure. The only way is usually to yank these versions and even that required changes to Bazel (https://github.com/bazelbuild/bazel/commit/2a2a4743ad624f7f238b7771b8ceedb7649e8f5c). We tried this in rules_go and now consider it a mistake.

Instead, if you really want to show this warning (we learned that almost nobody cared), I would recommend doing so from a module extension. That way, the warning is only shown if that version of the module is selected and actually used.

Reply:
I added this warning, but only did so because rules_go did so and so I thought it was best practice. Happy to remove it.

Reply:
Fix is in https://github.com/bazelbuild/rules_rust/pull/2345.